### DISCUSSION TITLE
Requesting: Switch component accessibility improvements.

### DISCUSSION THREAD

**Discussion by ispykenny** [heroui-inc/heroui#545] [Ideas]
Getting hit with a lighthouse score reduction for:

> When a toggle field doesn't have an accessible name, screen readers announce it with a generic name, making it unusable for users who rely on screen readers.

Would love a screen reader prop that is within (the first child element of) the `<label>` wrapper (but obviously not displayed to the UI).

This screen reader prop would simply take a string `type`

**Comment by jrgarciadev** [comment]
Hey @ispykenny thanks for the feedback, we have plans to migrate it to https://react-spectrum.adobe.com/react-aria/useSwitch.html as we have been doing with other components
